service-base/src/main/java/org/eclipse/hono/service/management/credentials/AbstractCredentialsManagementHttpEndpoint.java

```diff
@@ -0,0 +1,267 @@
+                    RegistryManagementConstants.CREDENTIALS_HTTP_ENDPOINT);
+
+    /**
+     * Creates an endpoint for a Vertx instance.
+     *
+     * @param vertx The Vertx instance to use.
+     * @throws NullPointerException if vertx is {@code null};
+     */
+    @Autowired
+    public AbstractCredentialsManagementHttpEndpoint(final Vertx vertx) {
+        super(Objects.requireNonNull(vertx));
+    }
+
+    /**
+     * Returns an empty String as this implementation does not use event bus.
+     */
+    @Override
+    protected String getEventBusAddress() {
+        return "";
+    }
+
+    @Override
+    public String getName() {
+        return CREDENTIALS_MANAGEMENT_ENDPOINT_NAME;
+    }
+
+    @Override
+    public void addRoutes(final Router router) {
+
+        final String pathWithTenantAndDeviceId = String.format("/%s/:%s/:%s",
+                getName(), PARAM_TENANT_ID, PARAM_DEVICE_ID);
+
+
+        // Add CORS handler
+        router.route(pathWithTenantAndDeviceId).handler(createCorsHandler(config.getCorsAllowedOrigin(), EnumSet.of(HttpMethod.GET, HttpMethod.PUT)));
+
+        final BodyHandler bodyHandler = BodyHandler.create();
+        bodyHandler.setBodyLimit(config.getMaxPayloadSize());
+
+        // get all credentials for a given device
+        router.get(pathWithTenantAndDeviceId).handler(this::getCredentialsForDevice);
+
+        // set credentials for a given device
+        router.put(pathWithTenantAndDeviceId).handler(bodyHandler);
+        router.put(pathWithTenantAndDeviceId).handler(this::extractRequiredJsonArrayPayload);
+        router.put(pathWithTenantAndDeviceId).handler(this::extractIfMatchVersionParam);
+        router.put(pathWithTenantAndDeviceId).handler(this::updateCredentials);
+    }
+
+
+    /**
+     * The service to forward requests to.
+     *
+     * @return The service to bind to, must never return {@code null}.
+     */
+    protected abstract CredentialsManagementService getService();
+
+    private void updateCredentials(final RoutingContext ctx) {
+
+        final Span span = Util.newChildSpan(SPAN_NAME_UPDATE_CREDENTIALS, TracingHandler.serverSpanContext(ctx), tracer,
+                getClass().getSimpleName());
+
+        final JsonArray credentials = ctx.get(KEY_REQUEST_BODY);
+
+        final String tenantId = getMandatoryRequestParam(PARAM_TENANT_ID, ctx, span);
+        final String deviceId = getMandatoryRequestParam(PARAM_DEVICE_ID, ctx, span);
+        final Optional<String> resourceVersion = Optional.ofNullable(getRequestParam(KEY_RESOURCE_VERSION, ctx, span, true));
+
+        if (credentials == null) {
+            final String msg = String.format("Missing request payload");
+            logger.debug(msg);
+            TracingHelper.logError(span, msg);
+            HttpUtils.badRequest(ctx, msg);
+            Tags.HTTP_STATUS.set(span, HttpURLConnection.HTTP_BAD_REQUEST);
+            span.finish();
+            return;
```

Comment:
I feel like I could extract this as `getMandatoryPayload()` in `AbstractHttpEndpoint`. @dejanb WDYT ?

Reply:
Yeah, looks like it